systems/controllers/QPCommon.cpp

```diff
@@ -86,13 +89,57 @@ std::shared_ptr<drake::lcmt_qp_controller_input> encodeQPInputLCM(const mxArray
+      msg->body_motion_data[i].in_floating_base_nullspace = static_cast<bool>(mxGetScalar(mxGetFieldSafe(body_motion_data, i, "in_floating_base_nullspace")));
+      msg->body_motion_data[i].control_pose_when_in_contact = static_cast<bool>(mxGetScalar(mxGetFieldSafe(body_motion_data, i, "control_pose_when_in_contact")));
+      const mxArray* quat_task_to_world = mxGetFieldSafe(body_motion_data, i, "quat_task_to_world"); 
+      if(mxGetM(quat_task_to_world) != 4 || mxGetN(quat_task_to_world) != 1)
+      {
+        mexErrMsgTxt("quat_task_to_world should be 4 x 1");
+      }
+      memcpy(msg->body_motion_data[i].quat_task_to_world,mxGetPr(quat_task_to_world),sizeof(double)*4);
+      const mxArray* translation_task_to_world = mxGetFieldSafe(body_motion_data, i, "translation_task_to_world");
+      if(mxGetM(translation_task_to_world) != 3 || mxGetN(translation_task_to_world) != 1)
+      {
+        mexErrMsgTxt("translation_task_to_world should be 3 x 1");
+      }
+      memcpy(msg->body_motion_data[i].translation_task_to_world, mxGetPr(translation_task_to_world),sizeof(double)*3);
+      const mxArray* xyz_kp_multiplier = mxGetFieldSafe(body_motion_data, i, "xyz_kp_multiplier");
+      if(mxGetM(xyz_kp_multiplier) != 3 || mxGetN(xyz_kp_multiplier) != 1)
+      {
+        mexErrMsgTxt("xyz_kp_multiplier should be 3 x 1");
+      }
+      memcpy(msg->body_motion_data[i].xyz_kp_multiplier, mxGetPr(xyz_kp_multiplier), sizeof(double)*3);
+      const mxArray* xyz_kd_multiplier = mxGetFieldSafe(body_motion_data, i, "xyz_kd_multiplier");
+      if(mxGetM(xyz_kd_multiplier) != 3 || mxGetN(xyz_kd_multiplier) != 1)
+      {
+        mexErrMsgTxt("xyz_kd_multiplier should be 3 x 1");
+      }
+      memcpy(msg->body_motion_data[i].xyz_kd_multiplier, mxGetPr(xyz_kd_multiplier), sizeof(double)*3);
+      const mxArray* expmap_kp_multiplier = mxGetFieldSafe(body_motion_data, i, "expmap_kp_multiplier");
+      if(mxGetM(expmap_kp_multiplier) != 1 || mxGetN(expmap_kp_multiplier) != 1)
+      {
+        mexErrMsgTxt("expmap_kp_multiplier should be 1 x 1");
+      }
+      msg->body_motion_data[i].expmap_kp_multiplier = mxGetScalar(expmap_kp_multiplier);
+      const mxArray* expmap_kd_multiplier = mxGetFieldSafe(body_motion_data, i, "expmap_kd_multiplier");
+      if(mxGetM(expmap_kd_multiplier) != 1 || mxGetN(expmap_kd_multiplier) != 1)
+      {
+        mexErrMsgTxt("expmap_kd_multiplier should be 1 x 1");
+      }
+      msg->body_motion_data[i].expmap_kd_multiplier = mxGetScalar(expmap_kd_multiplier);
+      const mxArray* weight_multiplier = mxGetFieldSafe(body_motion_data, i, "weight_multiplier");
+      if(mxGetM(weight_multiplier) != 6 || mxGetN(weight_multiplier) != 1)
+      {
+        mexErrMsgTxt("weight_multiplier should be 6 x 1");
+      }
+      memcpy(msg->body_motion_data[i].weight_multiplier, mxGetPr(weight_multiplier), sizeof(double)*6);
```

Comment:
use sizecheck from drakeUtil throughout this file. Possibly add a custom message argument to that function.